Sources/FoundationEssentials/Subprocess/Platforms/Subprocess+Unix.swift

```diff
@@ -0,0 +1,286 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Darwin) || canImport(Glibc)
+
+
+#if canImport(Darwin)
+import Darwin
+#elseif canImport(Glibc)
+import Glibc
+#endif
+
+import SystemPackage
+
+#if FOUNDATION_FRAMEWORK
+@_implementationOnly import _CShims
+#else
+package import _CShims
+#endif
+
+
+// MARK: - Signals
+extension Subprocess {
+    public struct Signal : Hashable, Sendable {
+        public let rawValue: Int32
```

Comment:
Do we really want to enforce `Int32` here or just make that an implementation detail? For most of our public APIs we prefer `Int` over specific `IntX` types.